src/tsfn.cpp

```diff
@@ -4,14 +4,25 @@
 #include "proj_internal.h"
 
 double pj_tsfn(double phi, double sinphi, double e) {
-    double denominator;
-    sinphi *= e;
+  /****************************************************************************
+   * Determine function ts(phi) defined in Snyder (1987), Eq. (7-10)
+   * Inputs:
+   *   phi = geographic latitude (radians)
+   *   e = eccentricity of the ellipsoid (dimensionless)
+   * Output:
+   *   ts = exp(-psi) where psi is the isometric latitude (dimensionless)
+   * Here isometric latitude is defined by
+   *   psi = log( tan(pi/4 + phi/2) *
+   *              ( (1 - e*sin(phi)) / (1 + e*sin(phi)) )^(e/2) )
+   *       = asinh(tan(phi)) - e * atanh(e * sin(phi))
+   ***************************************************************************/
 
-    /* avoid zero division, fail gracefully */
-    denominator = 1.0 + sinphi;
-    if (denominator == 0.0)
-        return HUGE_VAL;
-
-    return (tan (.5 * (M_HALFPI - phi)) /
-            pow((1. - sinphi) / (denominator), .5 * e));
+    double cosphi = cos(phi);
```

Comment:
if we need cos(phi) now then we should probably alter the signature of pj_tsfn to provide this value. This will enable callers to implicitly use sincos() to compute both sinp(hi) and cos(phi) faster than sin(phi) and cos(phi) in a isolated way

Reply:
I wonder about this.  I recommend putting a higher priority on removing the calls to tsfn and phi2 by reformulating the algorithms *not* to use Snyder's flakey `t = exp(-psi)` variable (which leads to oddities like the N and S poles being treated differently.